src/codeu/chat/client/maingui/MainGui.java

```diff
@@ -0,0 +1,77 @@
+package codeu.chat.client.maingui;
+
+import java.awt.*;
+import javax.swing.*;
+import javax.swing.border.Border;
+
+import codeu.chat.client.ClientContext;
+import codeu.chat.client.Controller;
+import codeu.chat.client.View;
+import codeu.chat.util.Logger;
+import codeu.chat.common.LoginInputCallback;
+
+// Chat - top-level client application - Java Simple GUI (using Java Swing)
+public final class MainGui implements LoginInputCallback{
```

Comment:
You may want to have something else implement LoginInputCallback -- in general it's cleaner to separate all the code handling your logic (onLoginRequest) with your GUI (all the layout / panels / event wiring) where possible.

Not something I think it's worth doing in this pull request, though, so just an optional suggestion for the future.